contracts/multisig-prover/src/encoding/bcs.rs

```diff
@@ -0,0 +1,206 @@
+use bcs::to_bytes;
+use cosmwasm_std::{HexBinary, Uint256};
+use itertools::Itertools;
+
+use crate::error::ContractError;
+
+use super::Data;
+
+pub fn command_params(
+    source_chain: String,
+    source_address: String,
+    destination_address: String,
+    payload_hash: HexBinary,
+) -> Result<HexBinary, ContractError> {
+    assert!(
+        destination_address.len() == 64,
+        "destination_address ({}) is not 32 bytes",
+        destination_address
+    );
+    let ret = to_bytes(&(
+        source_chain,
+        source_address,
+        <[u8; 32]>::try_from(&HexBinary::from_hex(&destination_address)?.to_vec()[..32])
+            .expect("couldn't convert destination_address to 32 byte array"),
+        payload_hash.to_vec(),
+    ))?;
+
+    Ok(ret.into())
+}
+
+// destination chain id must be u64 for sui
+fn chain_id_as_u64(chain_id: Uint256) -> u64 {
+    assert!(
+        chain_id <= Uint256::from(u64::MAX),
+        "chain_id ({}) is greater than u64 max",
+        chain_id
+    );
+    u64::from_le_bytes(
+        chain_id.to_le_bytes()[..8]
+            .try_into()
+            .expect("Couldn't convert u256 to u64"),
+    )
+}
+
+pub fn encode(data: &Data) -> HexBinary {
+    let destination_chain_id = chain_id_as_u64(data.destination_chain_id);
+
+    let (commands_ids, command_types, command_params): (Vec<[u8; 32]>, Vec<String>, Vec<Vec<u8>>) =
+        data.commands
+            .iter()
+            .map(|command| {
+                assert!(
```

Comment:
why is there an assert in production code?

Reply:
ignore this, apparently that is somewhat common practice in rust

Reply:
```suggestion
command_id
        .to_vec()
        .try_into()
        .expect("couldn't convert command id to 32 byte array")
```